src/proxy.rs

```diff
@@ -469,6 +477,104 @@ impl Dst for Uri {
     }
 }
 
+/// Get system proxies information.
+///
+/// It can only support Linux, Unix like, and windows system.  Note that it will always
+/// return a HashMap, even if something runs into error when find registry information in
+/// Windows system.  Note that invalid proxy url in the system setting will be ignored.
+///
+/// Returns:
+///     System proxies information as a hashmap like
+///     {"http": Url::parse("http://127.0.0.1:80"), "https": Url::parse("https://127.0.0.1:80")}
+pub fn get_proxies() -> HashMap<String, Url> {
+    let proxies: HashMap<String, Url> = get_from_environment();
+
+    if proxies.len() == 0 {
+        // don't cared if we can't get proxies from registry, just return an empty proxies.
+        #[cfg(target_os = "windows")]
+        let proxies = get_from_registry();
+        return proxies;
```

Comment:
```suggestion
        return get_from_registry();
```
no need for the variable

Reply:
Thanks for your comment!  I have write it in another way, then the variable is no needed.